server/api/contests.py

```diff
@@ -318,6 +344,24 @@ def create_or_update_all_contests(election: Election):
                 "Cannot update contests while batch tallies file is being processed."
             )
 
+        previous_contests = [
+            serialize_contest(contest) for contest in election.contests
+        ]
+
+    for contest in election.contests:
+        db_session.delete(contest)
```

Comment:
Why change this from the previous `Contest.query.filter_by(election_id=election.id).delete()`?

Reply:
Accessing `election.contests` on L348 just above causes the contests to be loaded into the ORM's cache, which caused a stale data error when later trying to delete and then commit the new contests. I honestly don't understand the error, but this fixed it.